tests/Utils/ObjectMixin.getMagicProperites().phpt

```diff
@@ -0,0 +1,117 @@
+<?php
+
+/**
+ * Test: Nette\Utils\ObjectMixin::getMagicProperties()
+ */
+
+use Nette\Utils\ObjectMixin;
+use Tester\Assert;
+
+
+require __DIR__ . '/../bootstrap.php';
+
+
+/**
+ * @property $getter
+ * @property $getter2 by ref
+ * @property $setter
+ * @property A\B $both with typehint
+ * @property $missing
+ * @property $Upper
+ * @property $cAse
+ * @property $private
+ * @property $protected
+ * @property $static
+ * @property-read $read
+ * @property-write $write
+ * @property-x $invalid
+ * abc @property $invalid2
+ */
+class TestClass
```

Comment:
* You test property annotations without types (except for one case), even though in the wild types are almost always present. 
* We may even consider not supporting property annotations without types (`mixed` can be used if type is unknown) to encourage programmers to write the annotations properly.
* You're missing edge case when `$this` is used as a type.